### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/get-started/command.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#743] [Announcements]
# https://theme-hope.vuejs.press/zh/get-started/command.html

本教程介绍 VuePress 项目命令。

https://theme-hope.vuejs.press/zh/get-started/command.html

**Comment by superen7527** [comment]
为什么 npx vp-update 以后 


❌ No VuePress bundler is detected in the current project! You should install one of @vuepress/bundler-vite, @vuepress/bundler-webpack
error The bundler or theme option is missing. For more details: https://v2.vuepress.vuejs.org/guide/troubleshooting.html#the-bundler-theme


不好意思 不懂编程。链接里面说的 和 网上找的 命令都不行 

npm install vite@latest 会出现更多问题。

npm i -D @vuepress/bundler-vite@next 还是一样的报错

这两个都没有效果。不知道怎么解决。

以为是我旧文件夹有问题，重新init一个项目，发现版本非常低(有没有基于最新版那种demo可以直接改src的)，尝试什么都不修改直接更新命令更新以后还是出现相同问题。很无奈。

**Reply by superen7527** [reply]
又找了几个命令。尝试 npm install @vuepress/bundler-vite --save-dev

然后 npm run docs:dev 还是这样 而且client好像不再支持了。
```
> docs@2.0.0 docs:dev
> vuepress dev src

@vuepress/client is no longer needed, you should remove it from deps and change all @vuepress/client imports to vuepress/client
error The bundler or theme option is missing. For more details: https://v2.vuepress.vuejs.org/guide/troubleshooting.html#the-bundler-theme-option-is-missing
```

**Reply by Mister-Hope** [reply]
add @latest with npm init.

**Reply by superen7527** [reply]
你好，感谢回复，通宵搞了一个晚上，没想到你轻描淡写一句话就能解决，哭T_T

最新版以后可以用更新命令了，以后的版本我如果还是更新失败，就再重新init新版本重新再创建一个demo然后迁移src好了，应该不会有问题了，就是不知道什么依赖会不会丢失 到时候再分析吧。感谢。

**Reply by Mister-Hope** [reply]
这个问题我也不是很理解而且我翻官方手册也没有看到。

就是npm不知道从哪个版本开始他会把这种init需要的create- 开头的包给缓存起来，你下次再运行的话，会优先使用你已经缓存的包。必须加at latest才能让他拉新包。

yarn和pnpm没有这个奇怪的行为

**Reply by Mister-Hope** [reply]
回头我改一下文档，把他们都加一下@latest

**Reply by superen7527** [reply]
好的好的，感谢关注。居然没人发现这个问题，看来就我一个人用npm而没用pnpm。

**Reply by superen7527** [reply]
还好我后来反应过来可能是npm特有问题，把命令带上了，不然直接说更新命令有问题，你要是以为是说的pnpm的更新命令有问题，但是又没问题，这个问题就要纠结一阵子了，哈哈。😂

**Comment by muchfish** [comment]
❌ No VuePress bundler is detected in the current project! You should install one of @vuepress/bundler-vite, @vuepress/bundler-webpack
error The bundler or theme option is missing. For more details
FUCK

**Comment by HuangGongWei** [comment]
为什么执行命令之后报这个，并且这个文件确实不在：
error Error: ENOENT: no such file or directory, lstat '/Users/gwh/Documents/yanAn/hgw/note/vuePress/gw-blog/src/.vuepress/config.ts.47438a8e.mjs'
